ohapi/utils_fs.py

```diff
@@ -168,6 +168,101 @@ def load_metadata_csv(input_filepath):
     return metadata
 
 
+def review_metadata_csv_project_specific(filedir, outputfilepath, subdirs,
+                                         max_bytes=MAX_FILE_DEFAULT):
+    """
+    Review metadata file for all files in a directory for a project.
+
+    :param filedir: This field is the filepath of the directory whose csv
+        has to be made.
+    :param outputfilepath: This field is the file path of the output csv.
+    :param subdirs: This field are the subdirectory of the given directory.
+    :param max_bytes: This field is the maximum file size to consider. Its
+        default value is 128m.
+    """
+    with open(outputfilepath, 'r') as outputfile:
```

Comment:
This and the other `review_metadata_csv` functions seem to replicate the data loading being done in `load_metadata_csv` functions. There shouldn't be this redundancy – data should be loaded from the files using the same code.

You can modify the `load_metadata` code if you think it could be improved. :smile: But I think the data loading bit needs to be non-duplicated. Then, the review function focuses on checking that data loads successfully, contains needed information, and that it's valid and formatted correctly – e.g. ISO 8601 date formatting – see this page for metadata requirements: https://www.openhumans.org/direct-sharing/on-site-data-upload/

Also: I know checking all potentially issues in validity and formatting could conceivably be quite extensive – so, rather than worry too much about catching *all* varieties of issues, can you describe what is being checked in the docstring for the function? (e.g. "checks for valid ISO 8601 format for `creation_date`, `start_date`, and `end_date` fields, if they are provided")